chartfx-math/src/main/java/de/gsi/math/MathGenBase.java

```diff
@@ -3132,391 +2624,92 @@ public static long LocationMinimum(short[] a, int length) {
-        final short[] temp = Sort(data, length, false);
-        if (length % 2 == 0) {
-            return (short) (0.5 * (temp[length / 2] + temp[length / 2 + 1]));
-        } else {
-            return temp[length / 2];
-        }
-    }
-
-    public static double Minimum(double[] data) {
-        return Minimum(data, data.length);
+    public static double minimum(double[] data) {
+        return minimum(data, data.length);
     }
 
     /**
      * @param data the input vector
      * @param length &lt;= data.length elements to be used
      * @return value of smallest vector element
      */
-    public static double Minimum(double[] data, int length) {
-        double val = +Double.MAX_VALUE;
+    public static double minimum(double[] data, int length) {
+        double val = +Double.MAX_VALUE; //// codegen: subst:int:Int:Integer
         for (int i = 0; i < length; i++) {
-            val = Math.min(val, data[i]);
-        }
-        return val;
-    }
-
-    public static float Minimum(float[] data) {
-        return Minimum(data, data.length);
-    }
-
-    /**
-     * @param data the input vector
-     * @param length &lt;= data.length elements to be used
-     * @return value of smallest vector element
-     */
-    public static float Minimum(float[] data, int length) {
-        float val = +Float.MAX_VALUE;
-        for (int i = 0; i < length; i++) {
-            val = Math.min(val, data[i]);
-        }
-        return val;
-    }
-
-    public static int Minimum(int[] data) {
-        return Minimum(data, data.length);
-    }
-
-    /**
-     * @param data the input vector
-     * @param length &lt;= data.length elements to be used
-     * @return value of smallest vector element
-     */
-    public static int Minimum(int[] data, int length) {
-        int val = +Integer.MAX_VALUE;
-        for (int i = 0; i < length; i++) {
-            val = Math.min(val, data[i]);
-        }
-        return val;
-    }
-
-    public static long Minimum(long[] data) {
-        return Minimum(data, data.length);
-    }
-
-    /**
-     * @param data the input vector
-     * @param length &lt;= data.length elements to be used
-     * @return value of smallest vector element
-     */
-    public static long Minimum(long[] data, int length) {
-        long val = +Long.MAX_VALUE;
-        for (int i = 0; i < length; i++) {
-            val = Math.min(val, data[i]);
-        }
-        return val;
-    }
-
-    public static short Minimum(short[] data) {
-        return Minimum(data, data.length);
-    }
-
-    /**
-     * @param data the input vector
-     * @param length &lt;= data.length elements to be used
-     * @return value of smallest vector element
-     */
-    public static short Minimum(short[] data, int length) {
-        short val = +Short.MAX_VALUE;
-        for (int i = 0; i < length; i++) {
-            if (val < data[i]) {
-                val = data[i];
-            }
+            val = java.lang.Math.min(val, data[i]);
```

Comment:
Codacy found an issue: [Unnecessary use of fully qualified name 'java.lang.Math.min' due to existing implicit import 'java.lang.*'](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=5780327)